vncviewer/MonitorArrangement.cxx

```diff
@@ -0,0 +1,414 @@
+      top_y = y;
+    }
+
+    if ((y + h) > bottom_y) {
+      bottom_y = y + h;
+    }
+
+    if (x < left_x) {
+      left_x = x;
+    }
+
+    if ((x + w) > right_x) {
+      right_x = x + w;
+    }
+  }
+
+  // Finally, get the dimensions of the monitor that we 
+  // are looking whether it is required. 
+  Fl::screen_xywh(x, y, w, h, m);
+
+  return inside(top_y, bottom_y, left_x, right_x, x, y) 
+      || inside(top_y, bottom_y, left_x, right_x, x+w, y)
+      || inside(top_y, bottom_y, left_x, right_x, x, y+h)
+      || inside(top_y, bottom_y, left_x, right_x, x+w, y+h);
+}
+
+double MonitorArrangement::scale()
+{
+  const int MARGIN = 20;
+  std::pair<int, int> size = this->size();
+
+  double s_w = static_cast<double>(this->w()-MARGIN) / static_cast<double>(size.first);
+  double s_h = static_cast<double>(this->h()-MARGIN) / static_cast<double>(size.second);
+
+  // Choose the one that scales the least, in order to
+  // maximize our use of the given bounding area.
+  if (s_w > s_h) {
+    return s_h;
+  } else {
+    return s_w;
+  }
+}
+
+std::pair<int, int> MonitorArrangement::size()
+{
+  int x, y, w, h;
+  int top, bottom, left, right;
+  int x_min, x_max, y_min, y_max;
+  x_min = x_max = y_min = y_max = 0;
+
+  for (int i = 0; i < Fl::screen_count(); i++) {
+    Fl::screen_xywh(x, y, w, h, i);
+    
+    top = y;
+    bottom = y + h;
+    left = x;
+    right = x + w;
+    
+    if (top < y_min) {
+      y_min = top;
+    }
+
+    if (bottom > y_max) {
+      y_max = bottom;
+    }
+
+    if (left < x_min) {
+      x_min = left;
+    }
+
+    if (right > x_max) {
+      x_max = right;
+    }
+  }
+  
+  return std::make_pair(x_max - x_min, y_max - y_min);
+}
+
+std::pair<int, int> MonitorArrangement::offset()
+{
+  double scale = this->scale();
+  std::pair<int, int> size = this->size();
+  std::pair<int, int> origin = this->origin();
+
+  int offset_x = (this->w()/2) - (size.first/2 * scale);
+  int offset_y = (this->h()/2) - (size.second/2 * scale);
+
+  return std::make_pair(offset_x + abs(origin.first), offset_y + abs(origin.second));
+}
+
+std::pair<int, int> MonitorArrangement::origin()
+{
+  int x, y, w, h, ox, oy;
+  ox = oy = 0;
+
+  for (int i = 0; i < Fl::screen_count(); i++) {
+    Fl::screen_xywh(x, y, w, h, i);
+
+    if (x < ox) {
+      ox = x;
+    }
+
+    if (y < oy) {
+      oy = y;
+    }
+  }
+
+  return std::make_pair(ox, oy);
+}
+
+bool MonitorArrangement::inside(
+  int top_y, int bottom_y, int left_x, int right_x, int x, int y)
+{
+  return (x > left_x) && (x < right_x) && (y > top_y) && (y < bottom_y);
+}
+
+std::string MonitorArrangement::description(int m)
+{
+  std::stringstream ss;
+  assert(m < (int) m_monitors.size());
+
+  // Get the platform specific name of the monitor. 
+  ss << name(m);
+
+  if (ss.str().empty()) {
+    int x, y, w, h;
+    Fl::screen_xywh(x, y, w, h, m);
+
+    // Fallback to showing resolution and position of monitor.
+    ss << w << "x" << h << "+" << x << "+" << y;
```

Comment:
The X11 geometry syntax is probably unfamiliar to most so I don't think it helps much here. Perhaps just resolution is sufficient?